talk/owt/sdk/ic/backgroundblur.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) <2021> Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "backgroundblur.h"
+
+#include <exception>
+#include <limits>
+
+#include "libyuv/convert.h"
+#include "opencv2/imgproc.hpp"
+#include "third_party/webrtc/api/video/i420_buffer.h"
+#include "third_party/webrtc/rtc_base/logging.h"
+
+#include "selfiesegmentationmodel.h"
+
+namespace owt {
+namespace ic {
+
+BackgroundBlur::BackgroundBlur(InferenceEngine::Core& core) : model(core) {}
+
+bool BackgroundBlur::SetParameter(const std::string& key,
+                                  const std::string& value) {
+  if (key == "model_path") {
+    return model.LoadModel(value);
+  } else if (key == "blur_radius") {
+    blur_radius_ = stoi(value);
+    return true;
+  }
+  return false;
+}
+
+rtc::scoped_refptr<webrtc::VideoFrameBuffer> BackgroundBlur::Process(
+    const rtc::scoped_refptr<webrtc::VideoFrameBuffer>& buffer) {
+  if (!model.IsLoaded()) {
+    RTC_LOG(LS_WARNING) << "Background blur model is not initialized.";
+    return buffer;
+  }
+
+  auto i420 = buffer->GetI420();

```

Comment:
You need to check if the buffer is a native handle or not. ToI420 may lead to crash if it's a native handle. For now you can reject processing for native handle.

Reply:
How to check native handle?

Reply:
webrtc::VideoFrameBuffer::type();